### ISSUE TITLE
pm2 shows process is online if it fails in "wait_ready" mode

### ISSUE THREAD

**Issue by Danoha** [Unitech/pm2#3147]
## What's going wrong?
I have two apps declared in my ecosystem.config.js but both are actually the same script.
The first app is declared with `wait_ready: true`, the second is not. Everything else is exactly the same.
Running `pm2 start ecosystem.config.js` gives following output:

```
[PM2][WARN] Applications app1, app2 not running, starting...
[PM2] App [app2] launched (1 instances)
[PM2] App [app1] launched (1 instances)
┌──────────┬────┬─────────┬─────┬─────────┬─────────┬────────┬─────┬────────┬────────┬──────────┐
│ App name │ id │ mode    │ pid │ status  │ restart │ uptime │ cpu │ mem    │ user   │ watching │
├──────────┼────┼─────────┼─────┼─────────┼─────────┼────────┼─────┼────────┼────────┼──────────┤
│ app1     │ 0  │ cluster │ 0   │ errored │ 2       │ 0      │ 0%  │ 0 B    │ danoha │ disabled │
│ app2     │ 1  │ cluster │ 0   │ errored │ 2       │ 0      │ 0%  │ 0 B    │ danoha │ disabled │
└──────────┴────┴─────────┴─────┴─────────┴─────────┴────────┴─────┴────────┴────────┴──────────┘
 Use `pm2 show <id|name>` to get more details about an app
```
Which is expected. But then I run `pm2 list` and saw this:

```
┌──────────┬────┬─────────┬─────┬─────────┬─────────┬────────┬─────┬────────┬────────┬──────────┐
│ App name │ id │ mode    │ pid │ status  │ restart │ uptime │ cpu │ mem    │ user   │ watching │
├──────────┼────┼─────────┼─────┼─────────┼─────────┼────────┼─────┼────────┼────────┼──────────┤
│ app1     │ 0  │ cluster │ 0   │ online  │ 2       │ 67s    │ 0%  │ 0 B    │ danoha │ disabled │
│ app2     │ 1  │ cluster │ 0   │ errored │ 2       │ 0      │ 0%  │ 0 B    │ danoha │ disabled │
└──────────┴────┴─────────┴─────┴─────────┴─────────┴────────┴─────┴────────┴────────┴──────────┘
 Use `pm2 show <id|name>` to get more details about an app
```
Which should show both apps errored.

## How could we reproduce this issue?

server.js

```
setTimeout(() => {
    console.error('something went wrong');
    process.exit(1);
}, 1000);
```

ecosystem.config.js

```
module.exports = {

	/**
	 * Application configuration section
	 * http://pm2.keymetrics.io/docs/usage/application-declaration/
	 */
	apps: [
		{
			name: 'app1',
			script: 'server.js',
			exec_mode: 'cluster',
			instances: 1,
			min_uptime: '10s',
			max_restarts: 3,
			wait_ready: true
		}, {
			name: 'app2',
			script: 'server.js',
			exec_mode: 'cluster',
			instances: 1,
			min_uptime: '10s',
			max_restarts: 3,
		},
	],
};
```

run

```bash
$ pm2 start ecosystem.config.js
$ pm2 list
```

## Supporting information

```
===============================================================================
--- PM2 REPORT (Mon Sep 11 2017 23:37:33 GMT+0200 (CEST)) ---------------------
===============================================================================
--- Daemon -------------------------------------------------
pm2d version         : 2.6.2
node version         : 8.4.0
node path            : /usr/bin/pm2
argv                 : /usr/bin/nodejs,/usr/lib/node_modules/pm2/lib/Daemon.js
argv0                : node
user                 : danoha
uid                  : 1000
gid                  : 1000
uptime               : 9min
===============================================================================
--- CLI ----------------------------------------------------
local pm2            : 2.6.2
node version         : 8.4.0
node path            : /usr/bin/pm2
argv                 : /usr/bin/nodejs,/usr/bin/pm2,report
argv0                : node
user                 : danoha
uid                  : 1000
gid                  : 1000
===============================================================================
--- System info --------------------------------------------
arch                 : x64
platform             : linux
type                 : Linux
cpus                 : Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz
cpus nb              : 4
freemem              : 8755507200
totalmem             : 16703430656
home                 : /home/danoha
===============================================================================
--- PM2 list -----------------------------------------------
┌──────────┬────┬─────────┬─────┬─────────┬─────────┬────────┬─────┬────────┬────────┬──────────┐
│ App name │ id │ mode    │ pid │ status  │ restart │ uptime │ cpu │ mem    │ user   │ watching │
├──────────┼────┼─────────┼─────┼─────────┼─────────┼────────┼─────┼────────┼────────┼──────────┤
│ app1     │ 0  │ cluster │ 0   │ online  │ 2       │ 5m     │ 0%  │ 0 B    │ danoha │ disabled │
│ app2     │ 1  │ cluster │ 0   │ errored │ 2       │ 0      │ 0%  │ 0 B    │ danoha │ disabled │
└──────────┴────┴─────────┴─────┴─────────┴─────────┴────────┴─────┴────────┴────────┴──────────┘
===============================================================================
--- Daemon logs --------------------------------------------
/home/danoha/.pm2/pm2.log last 20 lines:
PM2        | 2017-09-11 23:32:08: App [app2] with id [1] and pid [3534], exited with code [1] via signal [SIGINT]
PM2        | 2017-09-11 23:32:08: Starting execution sequence in -cluster mode- for app name:app2 id:1
PM2        | 2017-09-11 23:32:08: App name:app2 id:1 online
PM2        | 2017-09-11 23:32:08: App name:app2 id:1 disconnected
PM2        | 2017-09-11 23:32:08: App [app2] with id [1] and pid [3575], exited with code [1] via signal [SIGINT]
PM2        | 2017-09-11 23:32:08: Starting execution sequence in -cluster mode- for app name:app2 id:1
PM2        | 2017-09-11 23:32:08: App name:app1 id:0 disconnected
PM2        | 2017-09-11 23:32:08: App [app1] with id [0] and pid [3574], exited with code [1] via signal [SIGINT]
PM2        | 2017-09-11 23:32:08: Starting execution sequence in -cluster mode- for app name:app1 id:0
PM2        | 2017-09-11 23:32:09: App name:app2 id:1 online
PM2        | 2017-09-11 23:32:09: App name:app2 id:1 disconnected
PM2        | 2017-09-11 23:32:09: App [app2] with id [1] and pid [3615], exited with code [1] via signal [SIGINT]
PM2        | 2017-09-11 23:32:09: Script /home/danoha/Development/idos_web/server/index.js had too many unstable restarts (3). Stopped. "errored"
PM2        | 2017-09-11 23:32:09: App name:app1 id:0 disconnected
PM2        | 2017-09-11 23:32:09: App [app1] with id [0] and pid [3620], exited with code [1] via signal [SIGINT]
PM2        | 2017-09-11 23:32:09: Script /home/danoha/Development/idos_web/server/index.js had too many unstable restarts (3). Stopped. "errored"
PM2        | 2017-09-11 23:32:11: App name:app1 id:0 online
PM2        | 2017-09-11 23:32:11: (node:2552) [DEP0007] DeprecationWarning: worker.suicide is deprecated. Please use worker.exitedAfterDisconnect.
PM2        | 2017-09-11 23:32:11: App name:app1 id:0 online
PM2        | 2017-09-11 23:32:12: App name:app1 id:0 online

Please copy/paste the above report in your issue on https://github.com/Unitech/pm2/issues
```

Is it something I am doing wrong? Thank you very much!

**Comment by vahaknp** [comment]
+1!

**Comment by theoutlander** [comment]
This clearly seems like a bug. I've been going nuts trying to get this to work for the past hour!

**Comment by zhangda89** [comment]
DeprecationWarning: worker.suicide is deprecated. Please use worker.exitedAfterDisconnect.

should be update

**Comment by nsheaps** [comment]
+1

We use this for a quick health check on startup during our deploy, explicitly looking for any application that is not "online". This breaks that workflow and we now need to check it differently.

**Comment by lahdekorpi** [comment]
+1

This seems like a big issue in most cases. We are using deployment scripts that really need this.

A quick workaround is to get the JSON output and get the pid key. It's showing 0 on an errored process.

**Comment by kenjinsama** [comment]
+1

Get the JSON output works great, but this remains a big issue :(.

**Comment by bradisson** [comment]
+1

When timeout, it should set status as errored, not online.

**Comment by stgogm** [comment]
Removing and reinstalling the latest version seems to work for me:

`sudo npm rm -g pm2`
`sudo npm i -g pm2`

**Comment by yinrong** [comment]
+1 how to solve this? wait_ready seems useless

**Comment by JanGolle** [comment]
+1, still totally useless for cluster

**Comment by Unitech** [comment]
this has been fixed on 3.4.2

**Comment by JoshuaWise** [comment]
This has not been fixed in `3.4.2`. I tried that version and the most recent version, and all of them have this bug.

**Comment by adroste** [comment]
@Unitech Same here. Not fixed.
